README.md

```diff
@@ -84,6 +84,18 @@ int main()
 }
 ```
 
+Compatibility Guidelines
+========================
+
+The users if cppzmq are expected follow the guidelines below for breakage-free behaviour when upgrading cppzmq to newer versions (non-exhaustive list):
+
+* Do not depend on any macros defined in cppzmq unless explicitly exported.
```

Comment:
I would use a different term than `exported` here. Under `exported`, I'd expect some technical mechanism, but you simply mean that it is declared as such in the readme below.

What about saying:
```suggestion
* Do not depend on any macros defined in cppzmq unless explicitly declared public here.
```